### PR TITLE
refactor!: rename routeOptions to routeRules

### PR DESCRIPTION
**Pull request by danielroe**
### ❓ Type of change

- [ ] 📖 Documentation (updates to the documentation or readme)
- [ ] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [ ] 👌 Enhancement (improving an existing functionality like performance)
- [ ] ✨ New feature (a non-breaking change that adds functionality)
- [x] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description

This renames a number of route rule related utilities + config to use `route rules` rather than `route options`. This is a breaking change only for users tracking `nitropack-edge`.

### 📝 Checklist

- [x] I have linked an issue or discussion.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: src/runtime/route-options.ts [hunk 1]
```diff
@@ -2,35 +2,35 @@ import { eventHandler, H3Event, sendRedirect, setHeaders } from 'h3'
 import defu from 'defu'
```
Comment (pi0) [inline src/runtime/route-options.ts:2]: (we might rename file as well)

File: src/types/nitro.ts [hunk 2]
```diff
@@ -165,7 +165,7 @@ export interface NitroOptions extends PresetOptions {
   // Routing
   baseURL: string,
   handlers: NitroEventHandler[]
-  routes: { [path: string]: NitroRouteOptions }
+  routes: { [path: string]: NitroRouteRules }
```
Comment (pi0) [inline src/types/nitro.ts:168]: Do you mind also renaming `routes` (we can have backward compatibility for a while)

Comment (danielroe) [reply src/types/nitro.ts:168]: shall we also have config.nitro.routeRules or keep as config.nitro.routes?

Comment (danielroe) [reply src/types/nitro.ts:168]: I mean runtime config.

Comment (pi0) [reply src/types/nitro.ts:168]: Sure lets rename it!

**Pull request merged**